cmd/node_problem_detector.go

```diff
@@ -17,42 +17,21 @@ limitations under the License.
 package main
 
 import (
-	"net"
-	"net/http"
-	_ "net/http/pprof"
 	"os"
-	"strconv"
 
 	"github.com/golang/glog"
 	"github.com/spf13/pflag"
 
 	"k8s.io/node-problem-detector/cmd/options"
-	"k8s.io/node-problem-detector/pkg/custompluginmonitor"
-	"k8s.io/node-problem-detector/pkg/problemclient"
+	"k8s.io/node-problem-detector/pkg/exporters/k8sexporter"
+	"k8s.io/node-problem-detector/pkg/exporters/prometheusexporter"
+	"k8s.io/node-problem-detector/pkg/problemdaemon"
 	"k8s.io/node-problem-detector/pkg/problemdetector"
-	"k8s.io/node-problem-detector/pkg/systemlogmonitor"
 	"k8s.io/node-problem-detector/pkg/types"
 	"k8s.io/node-problem-detector/pkg/version"
-)
-
-func startHTTPServer(p problemdetector.ProblemDetector, npdo *options.NodeProblemDetectorOptions) {
-	// Add healthz http request handler. Always return ok now, add more health check
-	// logic in the future.
-	http.HandleFunc("/healthz", func(w http.ResponseWriter, r *http.Request) {
-		w.WriteHeader(http.StatusOK)
-		w.Write([]byte("ok"))
-	})
-	// Add the http handlers in problem detector.
-	p.RegisterHTTPHandlers()
 
-	addr := net.JoinHostPort(npdo.ServerAddress, strconv.Itoa(npdo.ServerPort))
-	go func() {
-		err := http.ListenAndServe(addr, nil)
-		if err != nil {
-			glog.Fatalf("Failed to start server: %v", err)
-		}
-	}()
-}
+	_ "k8s.io/node-problem-detector/pkg/systemstatsmonitor"
```

Comment:
Can we create a plugins.go in the main package, which just import plugins?

Like this https://github.com/containerd/containerd/blob/master/cmd/containerd/builtins_linux.go

Reply:
Sounds good :) Done.